### ISSUE TITLE
Doc comment missing when generating declarations for `const` methods from js files

### ISSUE THREAD

**Issue by karfau** [microsoft/TypeScript#35933]
**TypeScript Version:**  3.8.0-dev.20191228

**Search Terms:** declarations js missing jsdoc

**Code**

index.**js**(!):
```javascript
/**
 * const doc comment
 */
const x = (a) => {
    return '';
};

/**
 * function doc comment
 */
function b() {
    return 0;
}

module.exports = {x, b}
```
tsconfig.json:
```json
{
  "compilerOptions": {
    "target": "ES2015",
    "module": "none",
    "allowJs": true,
    "checkJs": true,
    "declaration": true,
    "outDir": "./",
    "strict": true,
    "noImplicitAny": false
  }
}
```

**Expected behavior:**
index.d.ts
```ts
/**
 * const doc comment
 */
export function x(a: any): string;
/**
 * second doc comment
 */
export function b(): number;
```

**Actual behavior:**
index.d.ts
```ts
export function x(a: any): string;
/**
 * second doc comment
 */
export function b(): number;
```

**Playground Link:** http://www.typescriptlang.org/play/?target=2&module=0&ts=3.8.0-dev.20191228&useJavaScript=true#code/PQKhCgAIUhbBPSATA9gY0mltYFMB2ALlCMOFvgM6GQAekAvJABQCGAlIwHyQDeUkQQCdchAK5D8kAOTSA3OAC+C8KAjRIlXBSTJ0mbHiIkyAMzH40hAJYopAI2ad+g4aIlSADAsXhwsFCQxABtcADpcWgAHFCFCSkY+WgAaSHtFIA

**Related Issues:** #22037 #5550

Current workaround: 
Converting those methods to the `function` syntax, solves the issue for now.

**Comment by jeanp413** [comment]
This is also fixed by #30253

**Comment by karfau** [comment]
Confirmed that it works in the nightly:
http://www.typescriptlang.org/play/?ts=Nightly#code/PQKhCgAIUhbBPSATA9gY0mltYFMB2ALlCMOFvgM6GQAekAvJABQCGAlIwHyQDeUkQQCdchAK5D8kAOTSA3OAC+C8KAjRIlXBSTJ0mbHiIkyAMzH40hAJYopAI2ad+g4aIlSADAsXhwsFCQxABtcADpcWgAHFCFCSkY+WgAaSHtFIA

@DanielRosenwasser 
Should this ticket be moved to the 3.8.0 milestone now?
Should this ticket be closed now?
